File: backend/plan_comptable.py
"""French Plan Comptable Général — charges (60-69) and produits (70-79)."""

PLAN_COMPTABLE_CHARGES = {
    "60": {
        "name": "Achats",
        "accounts": {
            "601": "Achats stockés - Matières premières et fournitures",
            "602": "Achats stockés - Autres approvisionnements",
            "603": "Variations de stocks",
            "604": "Achat d'études et prestations de services",
            "605": "Achats de matériels, équipements et travaux",
            "606": "Achats non stockés de matières et fournitures",
            "607": "Achats de marchandises",
            "608": "Frais accessoires d'achats",
            "609": "Rabais, remises et ristournes obtenus sur achats",
        },
    },
    "61": {
        "name": "Services extérieurs",
        "accounts": {
            "611": "Sous-traitance générale",
            "612": "Redevances de crédit-bail",
            "613": "Locations",
            "614": "Charges locatives et de co-propriété",
            "615": "Entretiens et réparations",
            "616": "Primes d'assurance",
            "617": "Études et recherches",
            "618": "Divers",
            "619": "RRR obtenus sur services extérieurs",
        },
    },
    "62": {
        "name": "Autres services extérieurs",
        "accounts": {
            "621": "Personnel extérieur à l'association",
            "622": "Rémunérations d'intermédiaires et honoraires",
            "623": "Publicité, publications, relations publiques",
            "624": "Transports de biens et transports collectifs du personnel",
            "625": "Déplacements, missions et réceptions",
            "626": "Frais postaux et de télécommunications",
            "627": "Services bancaires et assimilés",
            "628": "Divers",
            "629": "RRR obtenus sur autres services extérieurs",
        },
    },
    "63": {
        "name": "Impôts, taxes et versements assimilés",
        "accounts": {
            "631": "Impôts, taxes et versements sur rémunérations",
            "633": "Impôts, taxes et versements assimilés (autres)",
        },
    },
    "64": {
        "name": "Charges de personnel",
        "accounts": {
            "641": "Rémunérations du personnel",
            "645": "Charges de sécurité sociale et de prévoyance",
            "647": "Autres charges sociales",
            "648": "Autres charges de personnel",
            "649": "Remboursements de charges de personnel",
        },
    },
    "65": {
        "name": "Autres charges de gestion courante",
        "accounts": {
            "651": "Redevances pour concessions, brevets, licences, droits et valeurs similaires",
            "652": "Valeurs comptables des immobilisations incorporelles et corporelles cédées",
            "653": "Charges de la générosité du public",
            "654": "Pertes sur créances irrécouvrables",
            "655": "Quotes-parts de résultat sur opérations faites en commun",
            "657": "Aides financières",
            "658": "Pénalités et autres charges",
        },
    },
    "66": {
        "name": "Charges financières",
        "accounts": {
            "661": "Charges d'intérêts",
            "664": "Pertes sur créances rattachées à des participations",
            "665": "Escomptes accordés",
            "666": "Pertes de change",
            "667": "Charges nettes sur cessions de valeurs mobilières de placement",
            "668": "Autres charges financières",
        },
    },
    "67": {
        "name": "Charges exceptionnelles",
        "accounts": {
            "672": "Charges sur exercices antérieurs (à reclasser)",
            "678": "Autres charges exceptionnelles",
        },
    },
    "68": {
        "name": "Dotations aux amortissements, dépréciations et provisions",
        "accounts": {
            "681": "Dotations aux amortissements, dépréciations et provisions d'exploitation",
            "686": "Dotations aux amortissements, dépréciations et provisions financières",
            "687": "Dotations aux amortissements, dépréciations et provisions exceptionnelles",
        },
    },
    "69": {
        "name": "Impôts sur les bénéfices",
        "accounts": {
            "695": "Impôts sur les bénéfices",
            "699": "Produits de réduction d'impôts",
        },
    },
}

PLAN_COMPTABLE_PRODUITS = {
    "70": {
        "name": "Ventes de produits finis, prestations de services, marchandises",
        "accounts": {
            "701": "Ventes de produits finis",
            "702": "Ventes de produits intermédiaires",
            "703": "Ventes de produits résiduels",
            "704": "Travaux",
            "705": "Études",
            "706": "Prestations de services",
            "707": "Ventes de marchandises",
            "708": "Produits des activités annexes",
            "709": "RRR accordés par l'association",
        },
    },
    "71": {
        "name": "Production stockée (ou déstockage)",
        "accounts": {
            "713": "Variations de stocks",
        },
    },
    "72": {
        "name": "Production immobilisée",
        "accounts": {
            "721": "Immobilisations incorporelles",
            "722": "Immobilisations corporelles",
        },
    },
    "73": {
        "name": "Concours publics",
        "accounts": {
            "731": "Concours publics",
        },
    },
    "74": {
        "name": "Subventions d'exploitation",
        "accounts": {
            "741": "Subventions d'exploitation",
        },
    },
    "75": {
        "name": "Autres produits de gestion courante",
        "accounts": {
            "751": "Redevances pour concessions, licences, marques, procédés, droits et valeurs similaires",
            "752": "Revenus des immeubles non affectés aux activités de l'association",
            "753": "Versements des fondateurs ou consommation de la dotation",
            "754": "Ressources liées à la générosité du public",
            "755": "Contributions financières",
            "756": "Cotisations",
            "758": "Indemnités et autres produits",
        },
    },
    "76": {
        "name": "Produits financiers",
        "accounts": {
            "761": "Produits des participations",
            "762": "Produits des autres immobilisations financières",
            "763": "Revenus des autres créances",
            "764": "Revenus des valeurs mobilières de placement",
            "765": "Escomptes obtenus",
            "766": "Gains de change",
            "767": "Produits nets sur cession de valeurs mobilières de placement",
            "768": "Autres produits financiers",
        },
    },
    "77": {
        "name": "Produits exceptionnels",
        "accounts": {
            "772": "Produits sur exercices antérieurs (à reclasser)",
            "778": "Autres produits exceptionnels",
        },
    },
    "78": {
        "name": "Reprises sur amortissements, dépréciations, provisions et engagements",
        "accounts": {
            "781": "Reprises sur amortissements, dépréciations et provisions d'exploitation",
            "786": "Reprises sur amortissements, dépréciations et provisions financières",
            "787": "Reprises sur amortissements, dépréciations et provisions exceptionnelles",
        },
    },
    "79": {
        "name": "Transfert de charges",
        "accounts": {
            "791": "Transferts de charges d'exploitation",
            "796": "Transferts de charges financières",
            "797": "Transferts de charges exceptionnelles",
        },
    },
}

File: frontend/src/data/planComptable.js
// Plan Comptable Algérien — Classes 60-79

export const PLAN_COMPTABLE = [
  {
    code: "60",
    label: "Achats",
    type: "charge",
    subcomptes: [
      { code: "601", label: "Achats stockés - Matières premières et fournitures" },
      { code: "602", label: "Achats stockés - Autres approvisionnements" },
      { code: "603", label: "Variations de stocks" },
      { code: "604", label: "Achat d'études et prestations de services" },
      { code: "605", label: "Achats de matériels, équipements et travaux" },
      { code: "606", label: "Achats non stockés de matières et fournitures" },
      { code: "607", label: "Achats de marchandises" },
      { code: "608", label: "Frais accessoires d'achats" },
      { code: "609", label: "Rabais, remises et ristournes obtenus sur achats" },
    ],
  },
  {
    code: "61",
    label: "Services extérieurs",
    type: "charge",
    subcomptes: [
      { code: "611", label: "Sous-traitance générale" },
      { code: "612", label: "Redevances de crédit-bail" },
      { code: "613", label: "Locations" },
      { code: "614", label: "Charges locatives et de co-propriété" },
      { code: "615", label: "Entretiens et réparations" },
      { code: "616", label: "Primes d'assurance" },
      { code: "617", label: "Etudes et recherches" },
      { code: "618", label: "Divers" },
      { code: "619", label: "RRR obtenus sur services extérieurs" },
    ],
  },
  {
    code: "62",
    label: "Autres services extérieurs",
    type: "charge",
    subcomptes: [
      { code: "621", label: "Personnel extérieur à l'association" },
      { code: "622", label: "Rémunérations d'intermédiaires et honoraires" },
      { code: "623", label: "Publicité, publications, relations publiques" },
      { code: "624", label: "Transports de biens et transports collectifs du personnel" },
      { code: "625", label: "Déplacements, missions et réceptions" },
      { code: "626", label: "Frais postaux et de télécommunications" },
      { code: "627", label: "Services bancaires et assimilés" },
      { code: "628", label: "Divers" },
      { code: "629", label: "RRR obtenus sur autres services extérieurs" },
    ],
  },
  {
    code: "63",
    label: "Impôts, taxes et versements assimilés",
    type: "charge",
    subcomptes: [],
  },
  {
    code: "64",
    label: "Charges de personnel",
    type: "charge",
    subcomptes: [
      { code: "641", label: "Rémunérations du personnel" },
      { code: "645", label: "Charges de sécurité sociale et de prévoyance" },
      { code: "647", label: "Autres charges sociales" },
      { code: "648", label: "Autres charges de personnel" },
      { code: "649", label: "Remboursements de charges de personnel" },
    ],
  },
  {
    code: "65",
    label: "Autres charges de gestion courante",
    type: "charge",
    subcomptes: [
      { code: "651", label: "Redevances pour concessions, brevets, licences" },
      { code: "652", label: "Valeurs comptables des immobilisations cédées" },
      { code: "653", label: "Charges de la générosité du public" },
      { code: "654", label: "Pertes sur créances irrécouvrables" },
      { code: "655", label: "Quotes-parts de résultat sur opérations faites en commun" },
      { code: "657", label: "Aides financières" },
      { code: "658", label: "Pénalités et autres charges" },
    ],
  },
  {
    code: "66",
    label: "Charges financières",
    type: "charge",
    subcomptes: [
      { code: "661", label: "Charges d'intérêts" },
      { code: "664", label: "Pertes sur créances rattachées à des participations" },
      { code: "665", label: "Escomptes accordés" },
      { code: "666", label: "Pertes de change" },
      { code: "667", label: "Charges nettes sur cessions de valeurs mobilières de placement" },
      { code: "668", label: "Autres charges financières" },
    ],
  },
  {
    code: "67",
    label: "Charges exceptionnelles",
    type: "charge",
    subcomptes: [
      { code: "672", label: "Charges sur exercices antérieurs (à reclasser)" },
      { code: "678", label: "Autres charges exceptionnelles" },
    ],
  },
  {
    code: "68",
    label: "Dotations aux amortissements, dépréciations, provisions",
    type: "charge",
    subcomptes: [],
  },
  {
    code: "69",
    label: "Impôts sur les bénéfices",
    type: "charge",
    subcomptes: [],
  },
  {
    code: "70",
    label: "Ventes de produits finis, prestations, marchandises",
    type: "produit",
    subcomptes: [
      { code: "701", label: "Ventes de produits finis" },
      { code: "702", label: "Ventes de produits intermédiaires" },
      { code: "703", label: "Ventes de produits résiduels" },
      { code: "704", label: "Travaux" },
      { code: "705", label: "Etudes" },
      { code: "706", label: "Prestations de services" },
      { code: "707", label: "Ventes de marchandises" },
      { code: "708", label: "Produits des activités annexes" },
      { code: "709", label: "RRR accordés par l'association" },
    ],
  },
  {
    code: "71",
    label: "Production stockée (ou déstockage)",
    type: "produit",
    subcomptes: [
      { code: "713", label: "Variations de stocks" },
    ],
  },
  {
    code: "72",
    label: "Production immobilisée",
    type: "produit",
    subcomptes: [
      { code: "721", label: "Immobilisations incorporelles" },
      { code: "722", label: "Immobilisations corporelles" },
    ],
  },
  {
    code: "73",
    label: "Concours publics",
    type: "produit",
    subcomptes: [],
  },
  {
    code: "74",
    label: "Subventions d'exploitation",
    type: "produit",
    subcomptes: [],
  },
  {
    code: "75",
    label: "Autres produits de gestion courante",
    type: "produit",
    subcomptes: [
      { code: "751", label: "Redevances pour concessions, licences, marques" },
      { code: "752", label: "Revenus des immeubles non affectés aux activités" },
      { code: "753", label: "Versements des fondateurs ou consommation de la dotation" },
      { code: "754", label: "Ressources liées à la générosité du public" },
      { code: "755", label: "Contributions financières" },
      { code: "756", label: "Cotisations" },
      { code: "758", label: "Indemnités et autres produits" },
    ],
  },
  {
    code: "76",
    label: "Produits financiers",
    type: "produit",
    subcomptes: [
      { code: "761", label: "Produits des participations" },
      { code: "762", label: "Produits des autres immobilisations financières" },
      { code: "763", label: "Revenus des autres créances" },
      { code: "764", label: "Revenus des valeurs mobilières de placement" },
      { code: "765", label: "Escomptes obtenus" },
      { code: "766", label: "Gains de change" },
      { code: "767", label: "Produits nets sur cession de VMP" },
      { code: "768", label: "Autres produits financiers" },
    ],
  },
  {
    code: "77",
    label: "Produits exceptionnels",
    type: "produit",
    subcomptes: [
      { code: "772", label: "Produits sur exercices antérieurs (à reclasser)" },
      { code: "778", label: "Autres produits exceptionnels" },
    ],
  },
  {
    code: "78",
    label: "Reprises sur amortissements, dépréciations, provisions",
    type: "produit",
    subcomptes: [],
  },
  {
    code: "79",
    label: "Transfert de charges",
    type: "produit",
    subcomptes: [],
  },
];

export function inferJournalType(groupCode) {
  const g = parseInt(groupCode, 10);
  if (g >= 60 && g <= 62) return "ACHATS";
  if (g === 64) return "SALAIRES";
  if (g >= 70 && g <= 75) return "VENTES";
  if (g === 66 || g === 76) return "BANQUE";
  return "OPERATIONS_DIVERS";
}
